cmd/oras/internal/errors/errors.go

```diff
@@ -62,10 +62,18 @@ func CheckArgs(checker func(args []string) (bool, string), Usage string) cobra.P
 
 // NewErrEmptyTagOrDigest creates a new error based on the reference string.
 func NewErrEmptyTagOrDigest(ref registry.Reference) error {
-	return NewErrEmptyTagOrDigestStr(ref.String())
+	return NewErrEmptyTagOrDigestStr(ref.String(), nil, false)
 }
 
 // NewErrEmptyTagOrDigestStr creates a new error based on the reference string.
-func NewErrEmptyTagOrDigestStr(ref string) error {
-	return fmt.Errorf("%q: no tag or digest when expecting <name:tag|name@digest>", ref)
+func NewErrEmptyTagOrDigestStr(ref string, cmd *cobra.Command, needsTag bool) error {
```

Comment:
We can rename this function, removing the suffix `str` since it's not returning a string
```suggestion
func NewErrEmptyTagOrDigest(ref string, cmd *cobra.Command, needsTag bool) error {
```

Reply:
renamed.